lib/streamlit/DeltaGenerator.py

```diff
@@ -1604,10 +1604,12 @@ def button(self, element, label, key=None):
 
         """
         element.button.label = label
-        element.button.default = False
 
         ui_value = _get_widget_ui_value("button", element, user_key=key)
         current_value = ui_value if ui_value is not None else False
+
+        element.button.value = current_value
```

Comment:
Can you explain what this change is doing?

Reply:
Why is this necessary now when this behavior was working before? Maybe the bug was introduced somewhere else? Or was this never working in any previous version of streamlit?

Reply:
Doesn't the widget state manager (or something else) handle the interaction between the client state and server values? Not sure I'm understanding the basis behind this change. It should be unnecessary based on the state management code that we have.

Could this potentially be breaking some other case that we don't have tests for?

Reply:
We checked this with @tvst and it was happening also for previous versions, I don't know which version specifically started happening but I'll investigate it now.